src/EventGrid/EventGrid/Domain/GetAzureEventGridDomain.cs

```diff
@@ -70,10 +72,57 @@ public class GetAzureEventGridDomain : AzureEventGridCmdletBase
         [ValidateNotNullOrEmpty]
         public string ResourceId { get; set; }
 
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = EventGridConstants.ODataQueryHelp,
+            ParameterSetName = DomainNameParameterSet)]
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = EventGridConstants.ODataQueryHelp,
+            ParameterSetName = ResourceGroupNameParameterSet)]
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = EventGridConstants.ODataQueryHelp,
+            ParameterSetName = ResourceIdEventSubscriptionParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string ODataQuery { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = EventGridConstants.TopHelp,
+            ParameterSetName = DomainNameParameterSet)]
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = EventGridConstants.TopHelp,
+            ParameterSetName = ResourceGroupNameParameterSet)]
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = EventGridConstants.TopHelp,
+            ParameterSetName = ResourceIdEventSubscriptionParameterSet)]
+        [ValidateRange(1, 100)]
+        public int? Top { get; set; }
```

Comment:
@ahamad-MS nit: should this be nullable? If the user provides a value, it should be an integer between 1 and 100.

(this comment applies to all instances of this in the PR)

Reply:
Yes since we need to support no pagination scenario too (which is what previous customers are used to). In this context, if Top is null or not specified, then we will get all the results without pagination (old behavior which users are used to). 

If Top is specified, then the pagination behavior will kick in. 

If user specified Top as 0 or -negative  it will be an invalid value for pagination scenario and not a trigger for not supporting pagination.

Reply:
@ahamad-MS if the concern is about determining if the user provided a value to the `-Top` parameter, then you can do this two ways:

(1) The recommended way is to use `MyInvocation.BoundParameters.ContainsKey[nameof(Top)]` -- this is built into every cmdlet and will tell you whether or not the user provided a value for `-Top`. This will work if the parameter is nullable or not.

(2) If the parameter isn't nullable, then you can always check to see if the value is 0 -- even though the default value is outside of the range, PowerShell only does validation on the values provided by the user, so if you find the `-Top` is set to 0 in the cmdlet, then you know that the user didn't provide a value for it.

Reply:
OK, we are taking option 1 of using MyInvocation.BoundParameters.ContainsKey(nameof(Top)) and updating code and help text accordingly